pkg/server/streaming.go

```diff
@@ -44,18 +44,35 @@ func newStreamServer(c *criService, addr, port string) (streaming.Server, error)
 	}
 	config := streaming.DefaultConfig
 	config.Addr = net.JoinHostPort(addr, port)
-	runtime := newStreamRuntime(c)
-	if c.config.EnableTLSStreaming {
-		tlsCert, err := newTLSCert()
+	run := newStreamRuntime(c)
+	if !c.config.EnableTLSStreaming {
```

Comment:
Can we move all the validation logic into a small function? A bit messy to split around this function.

Reply:
Sure !
I choose to improve this by returning the type of listener instead of checking it again in `newStreamServer`.
I improved the error handling in this validation function and covered it by tests.